### ISSUE TITLE
net7.0 not supported for free static web app functions? 

### ISSUE THREAD

**Issue by paule96** [Azure/static-web-apps#1016]
A similar topic was discussed here: #996 

**Is your feature request related to a problem? Please describe.**

Currently I'm struggling with bringing my backend API of an Azure Static WebSite to dotnet 7.
For now I'm a bit confused in what is actually supported in the free tier.
First source is the [GA announcement](https://azure.microsoft.com/en-us/updates/generally-available-azure-static-web-apps-now-fully-supports-net-7/). It just says, dotnet 7 can be used.
Then I looked in the docs and found also [here ](https://learn.microsoft.com/en-us/azure/static-web-apps/apis-functions)that it should be fine as a managed function. But on the same site there is the heading [Constraints](https://learn.microsoft.com/en-us/azure/static-web-apps/apis-functions#constraints). And there the support for dotnet 7 is missing on the managed site.

The the question is now, what is when supported? 

**Describe the solution you'd like**

A clear description what frameworks / versions can be used in the managed functions, and if there are any exclusion of the free tier there, it should be noted somewhere.

**Comment by vivekjilla** [comment]
Hi @paule96 Thanks for the feedback. We will update the docs for `constraints` section.

.net 7.0 is support but only via isolated worker process (even by Azure Functions, this is the case.)
Same info and more details has been provided in this comment in the github issue you mentioned: https://github.com/Azure/static-web-apps/issues/996#issuecomment-1322575098

There is no difference between free tier vs standard tier wrt supported function versions.

Can you please try out .net7.0 using the isolated process project and see if it helps: https://learn.microsoft.com/en-us/azure/azure-functions/dotnet-isolated-process-guide

Please let me know if there are any other concerns. Thanks!

**Comment by paule96** [comment]
Hm okay. Then I maybe have another problem. Because my api with dotnet 7 seems to have startup problems. You can check it here:
https://kind-dune-004c2c103-test2.westeurope.azurestaticapps.net/api/GenerateSas

But there are no errors in the app insights. And I only have this problem with dotnet 7.

**Comment by vivekjilla** [comment]
@paule96 Looking at your build command, it could be due to the runtime identifier. 
Managed functions for dotnet are hosted on Windows for Static Web Apps, but I  see you providing  runtime as Linux . Can you please try changing that and see if it fixes the issue. Thanks!

**Comment by paule96** [comment]
okay I tried it now with the following options:

- `--runtime win-x64 --no-self-contained`
- `--runtime win-x86 --no-self-contained`
- `--runtime win-x64 --self-contained`
- `--runtime win-x86 --self-contained`

No option did work. Is it maybe not allowed to mix frameworks over environments?

So currently my production stage is on dotnet 6. (not isolated)
And my other environments like the environment `test2` are on dotnet 7 isolated.

**Comment by vivekjilla** [comment]
@paule96 Looking at this further, your function startup is failing with error `Did not find functions with language [dotnet-isolated]`. 

It could have something to do with the way the project is built but looking at your code I didn't find any apparent issue. 

Can you please try and see if you can run your project locally using Azure functions core tools and see if it helps figure out the issue: https://github.com/Azure/azure-functions-dotnet-worker#create-and-run-net-isolated-functions
or try to deploy the api to standalone function and see if it works. Thanks!

**Comment by paule96** [comment]
Hm if I publish my functions lokal with `dotnet publish -c Release ./src/CZ.Azure.FileExchange.Api/ -o temp/api` and then go the this directory and start the functions host with `func host start` it starts but with the following warnings:

```bash
Can't determine project language from files. Please use one of [--csharp, --javascript, --typescript, --java, --python, --powershell, --custom]
Can't determine project language from files. Please use one of [--csharp, --javascript, --typescript, --java, --python, --powershell, --custom]
Can't determine project language from files. Please use one of [--csharp, --javascript, --typescript, --java, --python, --powershell, --custom]
```

But it runs perfectly fine lokal.

The question is now, why not in the cloud. Also here the content of the worker and host json from the lokal publish dir:

worker.config.json

```json
{
  "description": {
    "language": "dotnet-isolated",
    "extensions": [ ".dll" ],
    "defaultExecutablePath": "dotnet",
    "defaultWorkerPath": "CZ.Azure.FileExchange.Api.dll",
    "workerIndexing": "false"
  }
}
```

host.json

```json
{
  "version": "2.0",
  "logging": {
    "applicationInsights": {
      "samplingSettings": {
        "isEnabled": true,
        "excludedTypes": "Request"
      }
    }
  }
}
```

**Comment by paule96** [comment]
@vivekjilla any news on this topic?

**Comment by Izzmo** [comment]
@paule96 I was able to get it working for me. What are you running into?

**Comment by vivekjilla** [comment]
@paule96 sorry I missed your original message. I will try to see if there are any other logs that might help here. 
By any chance, did you also try deploying to a function app directly and see if it's working? That might give some more insights as well.

**Comment by paule96** [comment]
Yes I tried it with azure functions. Surprisingly I get the following error:
<img width="907" alt="image" src="https://user-images.githubusercontent.com/11291885/217829853-777730b9-1e67-4092-81e2-45765fe430d6.png">

The problem is I don't know why.
So for somereason he can find locally all .dlls but not inside azure.

**Comment by paule96** [comment]
So I found more out. 

It really seems to be a problem with the `Environment` feature of static web apps.

So if you have a Static Webapp that looks like the following:

- `Production`
   - API DOTNET 6 **not** isolated 
- `PR 12`
   - API DOTNET 7 isolated

In this case your `PR 12` Environment will not work. It will always end in a 404 not found.

**Comment by paule96** [comment]
@vivekjilla are there any news on this?

**Comment by sachin-rathor** [comment]
Hi @paule96,

Thanks for sharing more details. We're looking into this issue and trying to repro and understand more. Will get back to you once we find more details.

**Comment by paule96** [comment]
@vivekjilla  @sachin-rathor any news here?

**Comment by paule96** [comment]
@sachin-rathor and @vivekjilla I have an update on this issue that maybe helps.

The previous information was wrong, in that this only will happen if the environments use different function frameworks.
It happens for me always if I have an environment that is:
- a dotnet 7 function
- is in isolated mode
- and is not in production.

So production works fine, but any other environment does not work.

**Comment by paule96** [comment]
Okay I found out now more. It depends where you build. My production slot is always build on windows. And my PR slots are always build on linux.
I changed my PR build now to also build on windows and it works... So I guess there is some difference in doining:

```bash
dotnet publish -c Release -o temp/api --runtime win-x86 --no-self-contained ./src/CZ.Azure.FileExchange.Api/CZ.Azure.FileExchange.Api.csproj
```

on windows and linux. I guess this is now more a problem of the azure functions team or?

what is your opinion @sachin-rathor & @vivekjilla
